### PR TITLE
Update Makefile to utilize go-1.8, update old dependencies

### PR DESCRIPTION
**Pull request by nskaggs**
## Description of change
This change updates the Makefile to depend on go-1.8 only for non gcc-go.

Why is this change needed? This change allows us to build with go-1.8, which will be required for new changes going forward.

## QA steps

The github-check-merge job is currently experimenting with building with go-1.8 to minimize impact. Once that job blesses this PR, we should be good to merge.

In addition, manual verification of running make install-dependencies.

This helps land https://bugs.launchpad.net/juju/+bug/1683099

### CODE REVIEW

**Comment by nskaggs** [comment]
Note, I'll be running test builds in here to verify this with CI.

**Comment by nskaggs** [comment]
!!build!!

**Comment by nskaggs** [comment]
!!build!!

**Comment by nskaggs** [comment]
!!build!!

**Comment by nskaggs** [comment]
!!build!!

**Comment by nskaggs** [comment]
!!build!!

**Comment by nskaggs** [comment]
!!build!!

**Comment by nskaggs** [comment]
!!build!!

**Comment by nskaggs** [comment]
!!build!!

File: Makefile [hunk 1]
```diff
@@ -98,14 +96,12 @@ rebuild-dependencies.tsv: godeps
 # Install packages required to develop Juju and run tests. The stable
 # PPA includes the required mongodb-server binaries.
 install-dependencies:
-ifeq ($(shell lsb_release -cs|sed -r 's/precise|wily/old/'),old)
-	@echo Adding juju PPAs for golang and mongodb-server
+	@echo Adding PPAs for golang and juju
```
Comment (wupeka) [inline Makefile]: I'd still clearly state that those are juju PPAs.

**Comment by nskaggs** [comment]
$$merge$$

**Comment by nskaggs** [comment]
!!build!!

**Comment by nskaggs** [comment]
$$merge$$

**Pull request merged**